### ISSUE TITLE
Steam for Linux doesn't always remember custom default directory.

### ISSUE THREAD

**Issue by IvoAndonov** [ValveSoftware/steam-for-linux#6781]
#### Your system information

* Steam client version (build number or date): 1576550254, steam api v020
* Distribution (e.g. Ubuntu): Arch Linux
* Opted into Steam client beta?: No
* Have you checked for system updates?: Every 3 hours, the system does that automatically

#### Please describe your issue in as much detail as possible:
On Linux I always use a custom directory for Steam which is /media/1000GB/STEAM and I've set it to be default.
The problem is that sometimes, usually after reboot, Steam forgets that directory. I have 2 games in that directory. But after rebooting and launching Steam, the program shows no games installed and I have to go to settings, remove the custom directory and then add it again.

Here's information about my sistem, if you need it:

[rado@arch]: ~>$ sysinfo
rado@arch
OS: Arch Linux
Kernel: x86_64 Linux 5.4.3-arch1-1
Uptime: 10m
Packages: 1099
Shell: bash
Resolution: 1920x1080
DE: Cinnamon 4.4.4
WM: Muffin
WM Theme: Mint-Y-Brownish_for-Cinnamon-v4_by-rado84 (Cinnamox-Rhino)
GTK Theme: Mint-Y-Brownish_for-Cinnamon-v4_by-rado84 [GTK2/3]
Icon Theme: Humanity-Dark
Font: Noto Sans Bold 10
Disk: 621G / 1,5T (44%)
CPU: Intel Core i7-4770 @ 8x 3.9GHz [33.0°C]
GPU: GeForce GTX 1050 Ti
RAM: 2347MiB / 32038MiB

Interestingly, that doesn't happen in Steam for Windows. There I also have a custom default directory and Steam for Windows remembers it all the time (D:\STEAM_WIN), no matter how many reboots I do.
The entire hard drive where the Steam for Linux default directory is automatically mounted at boot:
/dev/sdc1 /media/1000GB ntfs-3g defaults,auto,nofail,x-systemd.device-timeout=10,uid=1000,gid=1000,umask=000 0 0

I've also tried with /media/1000GB being an ext4 FS with all kinds of root rights for Steam but then Steam for Linux won't run any game from it and it rarely detects the games being located on that partition, so I switched it back to NTFS.

Since the last 3 updates of the linux kernel (5.4.4, 5.4.5 and 5.4.6) the problem I described above stopped happening which makes wonder is it possible that the problem was created by the linux kernel itself and not by Steam, but I'm posting this issue here cuz I can't be sure what the reason for it is.

#### Steps for reproducing this issue:

1. Set a custom default directory.
2. Install any game in the new directory (or simply cut and paste the game and the appmanifest to their respective locations in the new directory).
3. Exit Steam.
4. Reboot computer and open Steam.
5. The games inside the new default directory should be displayed as not installed.
6. If necessay keep rebooting until the games appear as not installed because it doesn't happen all the time.

**Comment by cakesale** [comment]
Hi there, just to confirm this is still no longer an issue on newer kernels?

Try mounting the drive to your home directory on an older kernel and see if that changes anything?

**Comment by IvoAndonov** [comment]
> Hi there, just to confirm this is still no longer an issue on newer kernels?
> 
> Try mounting the drive to your home directory on an older kernel and see if that changes anything?

That was na issue with the last few kernels, so yes, I'd say it did happen with older kernels. However, since kernel 5.4.3 was released it stopped happening but there were also several Steam updates, so I can't say for sure which fixed the problem - the kernel or the Steam updates. But that doesn't mean it won't start happening again, that's why I posted this bug.
